modules/plugin/geopkg/src/test/java/org/geotools/geopkg/GeoPkgDataStoreFactoryTest.java

```diff
@@ -57,4 +59,30 @@ public void testBaseDirectory() throws IOException {
 
         assertTrue(new File(tmp.getRoot(), "foo.gpkg").exists());
     }
+
+    @Test
+    public void testContentsOnly() throws IOException, SQLException {
+        Map<String, Serializable> map = new HashMap<>();
+        map.put(GeoPkgDataStoreFactory.CONTENTS_ONLY.key, "true");
+        map.put(GeoPkgDataStoreFactory.DBTYPE.key, "geopkg");
+        map.put(GeoPkgDataStoreFactory.DATABASE.key, "foo.gpkg");
+
+        GeoPkgDataStoreFactory factory = new GeoPkgDataStoreFactory();
+        factory.setBaseDirectory(tmp.getRoot());
+
+        // create some data to trigger file creation
+        SimpleFeatureTypeBuilder b = new SimpleFeatureTypeBuilder();
+        b.setName("foo");
+        b.setNamespaceURI("http://geotools.org");
+        b.setSRS("EPSG:4326");
+        b.add("geom", Point.class);
+        b.add("name", String.class);
+
+        JDBCDataStore data = factory.createDataStore(map);
+        data.createSchema(b.buildFeatureType());
+
+        assertTrue(
```

Comment:
Wouldn't this test pass even without providing the "contents_only" parameter?
A valid test must fail before the fix.
I was expecting to see the direct creation of a table, using direct SQL, that is, a table that would not be registered in gpkg_contents (createSchema should do that instead).

Reply:
done